openquake/hazardlib/mfd/truncated_gr.py

```diff
@@ -296,3 +296,16 @@ def from_moment(cls, min_mag, max_mag, bin_width, b_val, moment_rate):
         a_val = np.log10(rate)
         self = cls(min_mag, max_mag, bin_width, a_val, b_val)
         return self
+
+    @classmethod
+    def from_slip_rate(cls, min_mag, max_mag, bin_width, b_val,
+                       slip_rate, rigidity, area):
+        """
+        Calls .from_moment with moment = slip_rate * rigidity * area
+        """
+        mm = 1E-3  # conversion meters -> millimiters
```

Comment:
Here we work with SI hence we also need to convert rigidity and area accordingly
area *= 1e6 # km^2 to m^2
rigidity *= 1e9 # GPa to Pa
... unless we did it before :)

Reply:
That's your job to check ;-)